src/worker/plugins/loadPlugin.ts

```diff
@@ -94,3 +98,40 @@ export async function loadPlugin(server: PluginsServer, pluginConfig: PluginConf
     }
     return false
 }
+
+async function inferPluginCapabilities(server: PluginsServer, pluginConfig: PluginConfig) {
+    // infer on load implies there's no lazy loading, but all workers get
+    // these properties loaded
```

Comment:
We could lazy load them. We won't do anything with the loaded capabilities for the moment, and eventually set them in a separate "install" step anyway. No need to block for that.

Reply:
Hmm, I could unblock them, but that still doesn't mean the VMs are lazy loaded right? since `inferPluginCapabilities()` will load the VM on startup, before the worker gets any event.

Maybe I should just remove this comment, just realised it's confusing w.r.t what is being lazy loaded.

Reply:
The funny code here is this:

```ts
void pluginConfig.vm?.initialize!(server, pluginConfig, plugin.source, pluginDigest(plugin))
await inferPluginCapabilities(server, pluginConfig, prevConfig)
```

The first function is run in the background, the second one we await for. The second function awaits for a promise (`resolveInternalVm`) that gets set when the first function that's running in the background finishes.

In essence, the plugin is not lazy anymore, but we block until the VM setup has completed.

To get around it, we could just `void` the `inferPluginCapabilities` call and let them run in the background, however that also feels slightly hacky. 

A possibly better idea is to put this `inferPluginCapabilities` inside the `LazyVM.initialize` function. There you probably won't need `prevConfig` either. Just check if the current `pluginConfig.plugin.capabilities` matches reality or not. If not, update.

Reply:
that.. makes a lot of sense. Ha, would've been nice if I made this leap myself when thinking about the implementation.

Come to think of it, I don't need the `prevConfig` in the current setup either, for similar reasons - I already have a `pluginConfig.plugin.capabilities` to check.